pandas/core/internals.py

```diff
@@ -108,12 +108,7 @@ class Block(PandasObject):
     _concatenator = staticmethod(np.concatenate)
 
     def __init__(self, values, placement, ndim=None):
-        if ndim is None:
```

Comment:
cc @jbrockmendel. Pretty sure this is what you had in mind. Agreed it's cleaner.

Reply:
(needed a followup commit to use `self.ndim` a few lines down.